src/searchbar.cpp

```diff
@@ -114,6 +115,22 @@ void SearchBarLineEdit::hideSuggestions()
     m_completer.popup()->hide();
 }
 
+bool SearchBarLineEdit::eventFilter(QObject *, QEvent *event)
+{
+    if (!m_aboutToScrollPastEnd)
+        return false;
+
+    if (auto e = dynamic_cast<QKeyEvent *>(event))
+    {
+        if (e->key() == Qt::Key_PageDown && e->modifiers().testFlag(Qt::NoModifier))
```

Comment:
Now I wonder if it's better to handle key down here too (instead of your previous fix for it). Won't it eliminate the need to undo the handling of wrapping from bottom to top? If yes then the final code will be simpler, and so let's do it.

Reply:
So it seems the scrolling event from Key_down either reaches the popup first or isn't blocked from reaching the popup. I simply removed onScrollPastEnd and put the scrolling only inside the onScroll `!m_completer.popup()->currentIndex().isValid()` case.